services/spar/src/Spar/Scim/User.hs

```diff
@@ -745,54 +843,95 @@ synthesizeScimUser :: ST.ValidScimUser -> Scim.User ST.SparTag
 synthesizeScimUser info =
   let Handle userName = info ^. ST.vsuHandle
    in (Scim.empty ST.userSchemas userName (ST.ScimUserExtra (info ^. ST.vsuRichInfo)))
-        { Scim.externalId = Brig.renderValidExternalId $ info ^. ST.vsuExternalId,
+        { Scim.externalId = getOrCreateExternalId (info ^. ST.vsuAuthId),
           Scim.displayName = Just $ fromName (info ^. ST.vsuName),
-          Scim.active = Just . Scim.ScimBool $ info ^. ST.vsuActive
+          Scim.active = Just . Scim.ScimBool $ info ^. ST.vsuActive,
+          Scim.emails =
+            maybeToList $
+              authIdScimEmailWithSource (info ^. ST.vsuAuthId) >>= \ews ->
+                case ewsEmailSource ews of
+                  EmailFromExternalIdField -> Nothing
+                  EmailFromEmailsField -> emailToScimEmail (ewsEmail ews)
         }
+  where
+    getOrCreateExternalId :: AuthId -> Maybe Text
+    getOrCreateExternalId = \case
+      AuthSAML uref -> SAML.shortShowNameID (uref ^. SAML.uidSubject)
+      AuthSCIM (ScimDetails (ExternalId _ t) _) -> pure t
+      AuthBoth _ uref _ -> SAML.shortShowNameID (uref ^. SAML.uidSubject)
+
+    emailToScimEmail :: Email -> Maybe ScimEmail.Email
+    emailToScimEmail =
+      fmap scimEmail . toEmailAddress
```

Comment:
I think both the two copies of `unsafeEmailToScimEmail` in the tests and this `emailToScimEmail`, and `toEmailAddress` and `fromEmailAddress` can be made total: the two email types come from different packages (wire-api and email-validate), but they are both validated.

Anyway, that's a FUTUREWORK, and probably related to the question of whether `@handle` is also a valid email address (which it shouldn't be).

Reply:
https://wearezeta.atlassian.net/browse/SQSERVICES-379